pkg/proxy/healthcheck/service_health.go

```diff
@@ -59,20 +59,18 @@ type proxierHealthChecker interface {
 }
 
 func newServiceHealthServer(hostname string, recorder events.EventRecorder, listener listener, factory httpServerFactory, nodePortAddresses *utilproxy.NodePortAddresses, healthzServer proxierHealthChecker) ServiceHealthServer {
-
-	nodeAddresses, err := nodePortAddresses.GetNodeAddresses(utilproxy.RealNetwork{})
-	if err != nil || nodeAddresses.Len() == 0 {
-		klog.ErrorS(err, "Failed to get node ip address matching node port addresses, health check port will listen to all node addresses", "nodePortAddresses", nodePortAddresses)
-		nodeAddresses = sets.New[string]()
-		nodeAddresses.Insert(utilproxy.IPv4ZeroCIDR)
-	}
+	var nodeAddresses sets.Set[string]
```

Comment:
This is interesting, since we only get the nodeAddresses during creation, if new addresses are added to the node and we are not using the wildcard range, the healthcheck nodeport will not listen on those new addresses

Reply:
"pre-existing condition" :)

The health server is kind of a mess. See also #116486, #114702. And I just discovered yesterday that in `ipvs` mode, the "unhealthy timeout" is based on `--iptables-sync-period`, not `--ipvs-sync-period`...

Reply:
oh, so you already concluded that https://github.com/kubernetes/kubernetes/issues/114702#issuecomment-1384092803 😄